pkg/schedule/schedulers/balance_key_range.go

```diff
@@ -0,0 +1,151 @@
+package schedulers
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/unrolled/render"
+
+	"github.com/pingcap/log"
+
+	"github.com/tikv/pd/pkg/core"
+	"github.com/tikv/pd/pkg/core/constant"
+	"github.com/tikv/pd/pkg/errs"
+	sche "github.com/tikv/pd/pkg/schedule/core"
+	"github.com/tikv/pd/pkg/schedule/filter"
+	"github.com/tikv/pd/pkg/schedule/operator"
+	"github.com/tikv/pd/pkg/schedule/plan"
+	"github.com/tikv/pd/pkg/schedule/types"
+	"github.com/tikv/pd/pkg/utils/syncutil"
+)
+
+type balanceKeyRangeSchedulerHandler struct {
+	rd     *render.Render
+	config *balanceKeyRangeSchedulerConfig
+}
+
+func newBalanceKeyRangeHandler(conf *balanceKeyRangeSchedulerConfig) http.Handler {
+	handler := &balanceKeyRangeSchedulerHandler{
+		config: conf,
+		rd:     render.New(render.Options{IndentJSON: true}),
+	}
+	router := mux.NewRouter()
+	router.HandleFunc("/config", handler.updateConfig).Methods(http.MethodPost)
+	router.HandleFunc("/list", handler.listConfig).Methods(http.MethodGet)
+	return router
+}
+
+func (handler *balanceKeyRangeSchedulerHandler) updateConfig(w http.ResponseWriter, _ *http.Request) {
+	handler.rd.JSON(w, http.StatusBadRequest, "update config is not supported")
+}
+
+func (handler *balanceKeyRangeSchedulerHandler) listConfig(w http.ResponseWriter, _ *http.Request) {
+	conf := handler.config.clone()
+	if err := handler.rd.JSON(w, http.StatusOK, conf); err != nil {
+		log.Error("failed to marshal balance key range scheduler config", errs.ZapError(err))
+	}
+}
+
+type balanceKeyRangeSchedulerConfig struct {
+	syncutil.RWMutex
+	schedulerConfig
+	balanceKeyRangeSchedulerParam
+}
+
+type balanceKeyRangeSchedulerParam struct {
+	Role    string          `json:"role"`
+	Engine  string          `json:"engine"`
+	Timeout time.Duration   `json:"timeout"`
+	Ranges  []core.KeyRange `json:"ranges"`
```

Comment:
I wonder if we can let the API interface like: 

```
type KeyRange strut {
     alias string `json:"alias"`
     keyRange string `json:"range"`
}

```

Reply:
the key range may be some partition of  one table, and we can decode this key by ctl.

Reply:
The readability is too poor.

Reply:
understand,  I will add one property to record the table name.

Reply:
done